lfs.c

```diff
@@ -4396,6 +4396,32 @@ static int lfs_format_(lfs_t *lfs, const struct lfs_config *cfg) {
 }
 #endif
 
+struct lfs_tortoise_t
+{
+    lfs_block_t pair[2];
+    lfs_size_t i;
+    lfs_size_t period;
+};
+
+static int lfs_detect_cycles(lfs_mdir_t *dir, struct lfs_tortoise_t *tortoise)
```

Comment:
nit: This is an odd one, but the naming convention in littlefs is underscores only for "namespaces".

So one option: `lfs_tortoise_detectcycles`.

You can see other examples of this in the codebase.